Document Slack agent cross-repo access (#5541)
mintlify/docs: agent/index.mdx

- Update existing documentation for new features or API changes.
- Process and include images, diagrams, and other files from Slack attachments.
- Search and revise code examples and API references across your docs.

> - Reference source code from any repository where the Mintlify GitHub App is installed.

Reviewer: ```suggestion
- Reference source code from any repository that has the Mintlify GitHub App installed.
```
